kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/ClientCreator.java

```diff
@@ -34,15 +36,34 @@
     private InstanceProducer<KubernetesClient> producer;
 
     public void createClient(@Observes Configuration config) {
-        if (config.getMasterUrl() != null) {
+
+        if (config.getMasterUrl() == null) {
             producer.set(new DefaultKubernetesClient(new ConfigBuilder()
```

Comment:
Create `ConfigBuilder` outside `if` then depending on the values you populate the `ConfigBuilder`, i.e if(master!=null) then set masterUrl; if (token !=null) then set token;...

And then there is no if/else just ifs and at the end you construct the `DefaultKuebernetesClient` and inject in `InstanceProducer`